build/common.sh

```diff
@@ -950,6 +950,20 @@ function kube::release::package_kube_manifests_tarball() {
   mkdir -p "${dst_dir}/dns"
   tar c -C "${KUBE_ROOT}/cluster/saltbase/salt/kube-dns" ${objects} | tar x -C "${dst_dir}/dns"
 
+  # We leave the `{{ pillar['federations_domain_map'] }}` parameter as is, if
+  # the right federation environment variables isn't set. This is to allow
+  # users to provide these pillar values using the regular salt's mechanisms
+  # during cluster bootstrap.
+  if [[ "${FEDERATION:-}" == "true" ]]; then
+    FEDERATIONS_DOMAIN_MAP="${FEDERATIONS_DOMAIN_MAP:-}"
+    if [[ -z "${FEDERATIONS_DOMAIN_MAP}" && -n "${FEDERATION_NAME:-}" && -n "${DNS_ZONE_NAME:-}" ]]; then
```

Comment:
@nikhiljindal The operation that defaults `DNS_ZONE_NAME` (and `FEDERATION_NAME`) in `federation/cluster/common.sh` is independent of the operation here. This code here is part of the build and the operation that defaults `DNS_ZONE_NAME` is part of the deploy. We will have to default the same value twice if we want to default those values.

I don't like defaulting `DNS_ZONE_NAME` and `FEDERATION_NAME` here (as opposed to my general inclination towards reasonable defaults) for two reasons:

1. Doing that will entirely eliminate the users' ability to set those values later during cluster deployment, i.e. during the salt parameter substitution time.
2. It is not possible to get the e2e tests passing with the current defaults anyway and if a user wants to get their tests to pass they have to supply non-default values, i.e. the domain name that they own. So there is no reasonable default here.

However, for the automated tests, we will provide the `FEDERATIONS_DOMAIN_MAP` value directly to the Jenkins job. See https://github.com/kubernetes/test-infra/pull/211. So that should take care of the PR builder.